docs/Operators.md

```diff
@@ -7611,17 +7611,213 @@ Other versions of this operator: <a href="Changelog.md#GreaterOrEqual-12">12</a>
+
+The in-bounds coordinate vectors in `grid` are normalized to the range `[-1,1]`.
+The indices of `input` are in a range `[0,W)` and `[0,H)`, respectively, so we
+need to denormalize the coordinates in `grid` to put them into a range more
+appropriate for indexing into `input`. The calculation to do the denormalization
+depends on the value of `align_corners`.
+
+If `align_corners` is `0` (i.e. `false`), then for the horizontal (left-right)
+dimension, `-1` refers to the left border of the left-most column of pixels while
+`1` refers to the right border of the right-most column of pixels. The formula for
+denormalization of a coordinate `x` is then:
+
+  `denormalize(x, W, false) = ((x + 1) * W - 1) / 2` .
+
+If `align_corners` is `1` (i.e. `true`), then for the horizontal (left-right)
+dimension, `-1` refers to the center of the left-most column of pixels while `1`
+refers to the center of the right-most column of pixels. The formula for
+denormalization of a coordinate `x` is then:
+
+  `denormalize(x, W, true) = (x + 1) * (W - 1) / 2` .
+
+The normalized coordinates in `grid` are notated as `(x, y)` while the
+denormalized coordinates used for lookups are notated as `(row, col)`, matching
+PyTorch's GridSample. The concept of `x` is vertical (left-right) which
+corresponds to `col`, while `y` is horizontal (up-down) which corresponds to
+`row`. So in order to do a lookup in `input` based on `(x, y)`, it is also
+necessary to swap the two coordinates in addition to denormalization. All
+together, the coordinate calculation is then:
+
+  `(row, col) = (denormalize(y, H, align_corners), denormalize(x, W, align_corners))`
+
+where `y` goes first while in `grid` the ordering is `(x, y)` with `x` first. In
+other words, `grid` lists its coordinates in minor-to-major ordering (`y` and
+`row` are the major dimensions while `x` and `col` are minor). This is in
+contrast to e.g. `gather_nd` which uses the opposite ordering. So if `(-1, -1)`
+is the top-left corner, then `(-1, 1)` refers to the bottom-left corner, not the
+top-right corner.
+
+#### Interpolation of coordinates depending on `interpolation_mode`
+
+If the denormalized coordinates `(row, col)` are integer and in-bounds, then the
+value in the output at those coordinates is the value of `input` at those
+coordinates. If the denormalized coordinates are not integer, then the value is
+interpolated depending on the value of `interpolation_mode`.
+
+If `interpolation_mode` is `nearest`, then the value at those
+coordinates are the values in `input` of the nearest integer coordinates. This
+corresponds to the calculation
+
+`(row', col') = (round(row), round(col))` .
+
+If two integer points are equally close in a given dimension,
+corresponding to a coordinate ending in `.5`, then a rounding occurs
+according to the value of the nearest_mode attribute.
+
+For example, if `(row, col)=(1.5,3.5)`, then the coordinates used for the lookup
```

Comment:
I do not see explicit statement on this from pytorch doc either. in pytorch:
https://github.com/pytorch/pytorch/blob/cda0d5a57b9126c6d244fdd5b02198f05c742615/aten/src/ATen/native/GridSampler.cpp#L177
std::round is used. 
However in onnxruntime, it uses std::nearbyintf which depends on the environment. 
I perfer Pytorch's deterministic approach. 
@hwangdeyu  do you know why ORT uses std::nearbyintf instead of std::round?

Reply:
Let me know if I need to change this, but rounding towards nearest even makes sense to me as an unbiased rounding mode.